lib/elements/dropDown.js

```diff
@@ -30,66 +34,164 @@ class DropDown extends Element {
+    } else {
+      valuesToSelect = [values];
     }
 
     await doActionAwaitingNavigation(options, async () => {
       const { result } = await this.runtimeHandler.runtimeCallFunctionOn(selectBox, null, {
         objectId: this.get(),
-        arg: valueToSelect,
+        arg: valuesToSelect,
         returnByValue: true,
       });
       const { stack, error } = result.value;
       if (stack) {
         throw new Error(error + '\n' + stack);
+      } else if (error) {
+        throw new Error(error);
       }
-      if (result.value.isDisabled) {
-        throw new Error(`Cannot set value ${value} on a disabled field`);
-      }
-      if (!result.value.isAvailable) {
-        throw new Error(`Option ${value} not available in drop down`);
+      let disabledValues = [];
+      let unavailableValues = [];
+      Object.keys(result.value).forEach((key) => {
+        if (result.value[key].isDisabled) {
+          disabledValues.push(key);
+        }
+        if (!result.value[key].isAvailable) {
+          unavailableValues.push(key);
+        }
+      });
+
+      if (disabledValues.length > 0) {
+        throw new Error(`Cannot set value ${disabledValues.join(',')} on a disabled field`);
+      } else if (Object.keys(result.value).length === 0) {
+        if (typeof values === 'undefined') {
+          throw new Error('Option undefined not available in drop down');
+        } else if (!values) {
+          throw new Error(`Option ${values} not available in drop down`);
+        } else if ('index' in values) {
+          throw new Error(
+            `Option ${
+              Array.isArray(values.index) ? values.index.join(',') : values.index
+            } not available in drop down`,
+          );
+        } else if (Array.isArray(values)) {
+          throw new Error(
+            `Option ${values
+              .map((value) =>
+                typeof value === 'undefined' ? 'undefined' : !value ? 'null' : value,
+              )
+              .join(',')} not available in drop down`,
+          );
+        }
+      } else if (unavailableValues.length > 0) {
+        throw new Error(`Option ${unavailableValues.join(',')} not available in drop down`);
       }
     });
-    descEvent.emit('success', 'Selected ' + (value.index || value));
+
+    descEvent.emit('success', 'Selected ' + (values.index || values));
   }
+
   async value() {
     function getValue() {
-      if (this.value) {
-        return this.value;
+      let selectedValues = [];
+      for (let index = 0; index < this.options.length; index++) {
+        if (this.options[index].selected || this.selectedIndex === index) {
+          if (this.options[index].value) {
+            selectedValues.push(this.options[index].value);
+          } else {
+            selectedValues.push(this.options[index].innerText);
+          }
+        }
       }
-      return this.innerText;
+      return selectedValues.join(',');
```

Comment:
Why can't we just return the array `selectedValues `?

Reply:
If we just return array, it will not be backward compatible for which single value was provided.

Reply:
Thought - Should we have a new function called `values()` which would return multiple values in array format. 
@NivedhaSenthil @zabil

Reply:
> Thought - Should we have a new function called values()

You mean selected values right? Yeah, that makes sense. Can be part of another PR. @saikrishna321 can you create an issue for that?

Reply:
@zabil Yes, for selected values. Will create a new issue for that.

Reply:
@luthrap We can address this here  https://github.com/getgauge/taiko/issues/1613. Can this be reverted?

Reply:
Sure, will revert and update the test cases. Meanwhile, it appears that a lot of extra files have been checked in as part of the PR. This happened while signing off the previous commits, I mistakenly signed in those commits that should not belong to this PR. Could it be removed from here or shall i open a new clean PR request.

Reply:
> Sure, will revert and update the test cases.

Sorry, if it's been already implemented let's keep it! Maybe refer to the issue in this PR

> Could it be removed from here or shall i open a new clean PR request.

I noticed that too. You could update the same PR by fixing the merge and arranging the commits.

Reply:
>if it's been already implemented let's keep it

@zabil It's done as a part of current function `value()`

Reply:
We can move the value() code of the PR request in the values() function and have the original logic in the value() function. This will require a change in a couple of test cases and would conclude the PR correctly.

Reply:
> It's done as a part of current function value()

IMHO that should be fine. It might be confusing to have two different API's to return values depending on the type of the box when there is a common API to set the values depending on the type of the combo box.

So I am assuming `value` returns and array when there are multiple values selected

@saikrishna321 luthrap What do you think?

Reply:
Agreed, we should be encapsulating the logic in one function only.
```
if(selectedValues.length <= 1){
  return selectedValues.join()
else{
  return selectedValues
}
```

Reply:
> So I am assuming value returns an array when there are multiple values selected

Agree!